### ISSUE TITLE
comment_references false positive due to markdown link syntax

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/linter#5017]
I'm trying to clean up the SDK's `_fe_analyzer_shared` package to satisfy the `comment_references` lint, and I'm currently getting a false positive from the following comment (from `_fe_analyzer_shared/lib/src/base/errors.dart`):

```dart
/// An error code associated with an `AnalysisError`.
///
/// Generally, messages should follow the [Guide for Writing
/// Diagnostics](https://github.com/dart-lang/sdk/blob/main/pkg/front_end/lib/src/base/diagnostics.md).
```

I think the lint needs to consider the `[...](...)` markdown link syntax to be valid regardless of the text within `[...]`.

**Comment by srawlins** [comment]
What version of Dart were you using to run the lint rule? I ask because I think I fixed this in https://dart-review.googlesource.com/c/sdk/+/374122, submitted as eb57d21bf36eb1dca27b93f35ac841f18d43614f.

**Comment by srawlins** [comment]
Closing as duplicate, but if you reproduce with a version of Dart built in the last week, please re-open.

**Comment by stereotype441** [comment]
> Closing as duplicate, but if you reproduce with a version of Dart built in the last week, please re-open.

Confirmed that this problem no longer occurs. Thanks, Sam!
